src/full_automatic.rs

```diff
@@ -110,6 +109,9 @@ struct CollectNonReferenceArgTypes {
 }
 
 impl<'ast> Visit<'ast> for CollectNonReferenceArgTypes {
+    fn visit_receiver(&mut self, _: &'ast syn::Receiver) {
+        // Do nothing: explicitly ignore any receiver type.
+    }
```

Comment:
Without this, the built-in `visit_receiver` is called which then calls `visit_type` on the receiver type. Syn 1.0 didn't call visit_type on receivers (I'm a little confused because the receiver type is implicit, but it seems to be represented as if the type is explicit when parsing).